api/src/routes/crud.ts

```diff
@@ -501,21 +501,91 @@ router.post("/medication-request", [requireJWTMiddleware], async (req: Request,
                 }
             
             } else {
-                console.log(`Error: property ${medication} not found`);                
+                console.log(`Error: property ${medication} not found`);
             }
 
             const { siteCoding, routeCoding, methodCoding, unitCoding } = administrationCodings;
-            const medicationRequestDetails = MedicationRequest({ id, ...req.body[medication] }, medicationCoding, siteCoding, routeCoding, methodCoding, unitCoding);    
+            const medicationRequestDetails = MedicationRequest({ id, ...req.body[medication] }, medicationCoding, siteCoding, routeCoding, methodCoding, unitCoding);
             
-            await FhirApi({ url: `/MedicationRequest/${id}`, method: "PUT", data: JSON.stringify(medicationRequestDetails) });              
+            await FhirApi({ url: `/MedicationRequest/${id}`, method: "PUT", data: JSON.stringify(medicationRequestDetails) });
             
             medicationRequestIds.push({ medication, status: "success", medicationRequestId: id });
-        }     
+        }
         
-        res.json({ status: "success", data: medicationRequestIds });                    
+        res.json({ status: "success", data: medicationRequestIds });
     } catch (err) {
         res.json({ err, status: "error" });
     }
-})
+});
+
+router.post("/allergy-intolerance", [requireJWTMiddleware], async (req: Request, res: Response) => {
+    
+    try {
+
+        let id = uuidv4();
+
+        const { patientId, vaccine } = req.body;
+
+        const getVaccineCoding = () => {
+
+            if (Object.keys(medicalCodes).indexOf(vaccine) > -1) {            
+                return ({ status: "success", data: medicalCodes[vaccine] });                
+            } else {
+                return ({status:"error", data:`Error: property ${vaccine} not found`});
+            }
+        }
+
+        const { status, data } = getVaccineCoding();
+
+        if (status !== "success") {
+            res.statusCode = 400;
+            res.json(data);            
+            return;
+        }
+
+        const vaccineCode = data.split(":")[1];
+
+        // retrieve last administered vaccine matching the suspected cause substance for given child
+        let lastSimilarSubstance = await (await FhirApi({ url: `/Immunization?patient=${patientId}&vaccine-code=${vaccineCode}&_sort=-_lastUpdated`, method: "GET" })).data;
```

Comment:
What is the expected structure of the lastSimilarSubstance response from the FHIR API? Are there safeguards against unexpected API changes?